pkg/cluster/internal/create/actions/config/config.go

```diff
@@ -62,39 +65,66 @@ func (a *Action) Execute(ctx *actions.ActionContext) error {
 		return errors.Wrap(err, "failed to get kubernetes version from node")
 	}
 
-	// get the control plane endpoint, in case the cluster has an external load balancer in
-	// front of the control-plane nodes
+	// get the control plane endpoint
 	controlPlaneEndpoint, err := nodes.GetControlPlaneEndpoint(allNodes)
 	if err != nil {
 		// TODO(bentheelder): logging here
 		return err
 	}
 
-	// get kubeadm config content
-	kubeadmConfig, err := getKubeadmConfig(
-		ctx.Config,
-		kubeadm.ConfigData{
-			ClusterName:          ctx.ClusterContext.Name(),
-			KubernetesVersion:    kubeVersion,
-			ControlPlaneEndpoint: controlPlaneEndpoint,
-			APIBindPort:          kubeadm.APIServerPort,
-			APIServerAddress:     ctx.Config.Networking.APIServerAddress,
-			Token:                kubeadm.Token,
-			PodSubnet:            ctx.Config.Networking.PodSubnet,
-		},
-	)
+	// create kubeadm init config
+	fns := []func() error{}
+
+	kubeadmClusterConfig := kubeadm.ConfigData{
+		ClusterName:          ctx.ClusterContext.Name(),
+		KubernetesVersion:    kubeVersion,
+		ControlPlaneEndpoint: controlPlaneEndpoint,
+		APIBindPort:          kubeadm.APIServerPort,
+		APIServerAddress:     ctx.Config.Networking.APIServerAddress,
+		Token:                kubeadm.Token,
+		PodSubnet:            ctx.Config.Networking.PodSubnet,
+		ControlPlane:         true,
+	}
+
+	fns = append(fns, func() error {
+		return writeKubeadmConfig(ctx.Config, kubeadmClusterConfig, node)
+	})
 
+	// create the kubeadm join configuration for secondary control plane nodes if any
+	secondaryControlPlanes, err := nodes.SecondaryControlPlaneNodes(allNodes)
```

Comment:
-1
Each node should get it's own config file but they should come from the same template and inputs except for the node IP.

This needs to be consistent for patching, and so that nodes with different Kubernetes versions work.

Reply:
yeah, that´s what I´ve changed from the first version.
I think that now the configs and templates are the same for all nodes, all of them are generated with the `writeKubeadmConfig` function, the only difference between nodes config are the node IP address and the isControlPlane flag

Reply:
thanks! :-)